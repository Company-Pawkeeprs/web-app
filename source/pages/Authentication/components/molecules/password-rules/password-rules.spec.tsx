import { render, screen } from '@testing-library/react';
import { describe, expect, it, } from 'vitest';

import PasswordRules from './password-rules';

describe('PasswordRules component', () => {

    it('should match snapshot (Unit)', () => {
        const { container } = render(<PasswordRules value="abcd1234" />)

        expect(container).toMatchSnapshot()
    })

    it('renders password rules list (Unit)', () => {
        render(<PasswordRules value="" />);
        const passLength = screen.getByText(/Mínimo 8 caracteres/i);
        expect(passLength).toBeInTheDocument();
        const passLower = screen.getByText(/Uma letra minúscula/i);
        expect(passLower).toBeInTheDocument();
        const passUpper = screen.getByText(/Uma letra maiúscula/i);
        expect(passUpper).toBeInTheDocument();
        const passNumber = screen.getByText(/Um número/i);
        expect(passNumber).toBeInTheDocument();
    });

    it('validates password length (Unit)', () => {
        render(<PasswordRules value="test" />);
        const passLength = screen.getByText(/Mínimo 8 caracteres/i);
        expect(passLength).toHaveClass('text-danger');
    });

    it('validates lowercase letters in password (Unit)', () => {
        render(<PasswordRules value="TEST1234" />);
        const passLower = screen.getByText(/Uma letra minúscula/i);
        expect(passLower).toHaveClass('text-danger');
    });

    it('validates uppercase letters in password (Unit)', () => {
        render(<PasswordRules value="test1234" />);
        const passUpper = screen.getByText(/Uma letra maiúscula/i);
        expect(passUpper).toHaveClass('text-danger');
    });

    it('validates numbers in password (Unit)', () => {
        render(<PasswordRules value="testTEST" />);
        const passNumber = screen.getByText(/Um número/i);
        expect(passNumber).toHaveClass('text-danger');
    });

    it('validates password meets all requirements (Unit)', () => {
        render(<PasswordRules value="testTEST1234" />);
        const passLength = screen.getByText(/Mínimo 8 caracteres/i);
        expect(passLength).toHaveClass('text-success');
        const passLower = screen.getByText(/Uma letra minúscula/i);
        expect(passLower).toHaveClass('text-success');
        const passUpper = screen.getByText(/Uma letra maiúscula/i);
        expect(passUpper).toHaveClass('text-success');
        const passNumber = screen.getByText(/Um número/i);
        expect(passNumber).toHaveClass('text-success');
    });

    it('should render correctly (Unit)', () => {
        render(<PasswordRules value="" />);
        expect(screen.getByRole('heading', { level: 4 })).toHaveTextContent('Senha deve conter:');
        expect(screen.getByRole('list')).toBeInTheDocument();
    });

    it('should display the correct password rules based on the input value (Unit)', () => {
        const { rerender } = render(<PasswordRules value="" />);

        const passLength = screen.getByTestId(/pass-length/);
        const passLower = screen.getByTestId(/pass-lower/);
        const passUpper = screen.getByTestId(/pass-upper/);
        const passNumber = screen.getByTestId(/pass-number/);

        expect(passLength).toHaveClass('text-danger');
        expect(passLower).toHaveClass('text-danger');
        expect(passUpper).toHaveClass('text-danger');
        expect(passNumber).toHaveClass('text-danger');

        rerender(<PasswordRules value="senhaValida1" />);

        expect(screen.getByText('Mínimo 8 caracteres')).toHaveClass('text-success');
        expect(screen.getByText('Uma letra minúscula (a-z)')).toHaveClass('text-success');
        expect(screen.getByText('Uma letra maiúscula (A-Z)')).toHaveClass('text-success');
        expect(screen.getByText('Um número (0-9)')).toHaveClass('text-success');
    });
});
