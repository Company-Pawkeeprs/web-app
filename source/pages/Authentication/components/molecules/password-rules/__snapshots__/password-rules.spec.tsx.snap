// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PasswordRules component > should match snapshot (Unit) 1`] = `
<div>
  <div
    class="card p-3 bg-light rounded shadow-none"
  >
    <h4
      class="fs-11"
    >
      Senha deve conter:
    </h4>
    <ul
      class="list-group list-group-flush"
    >
      <li
        class="list-group-item fs-10 border-0 text-success"
        data-testid="pass-length"
        id="pass-length"
      >
        <i
          class="align-middle me-1 ri-check-line"
        />
        Mínimo 8 caracteres
      </li>
      <li
        class="list-group-item fs-10 border-0 text-success"
        data-testid="pass-lower"
        id="pass-lower"
      >
        <i
          class="align-middle me-1 ri-check-line"
        />
        Uma letra minúscula (a-z)
      </li>
      <li
        class="list-group-item fs-10 border-0 text-danger"
        data-testid="pass-upper"
        id="pass-upper"
      >
        <i
          class="align-middle me-1 ri-close-line"
        />
        Uma letra maiúscula (A-Z)
      </li>
      <li
        class="list-group-item fs-10 border-0 text-success"
        data-testid="pass-number"
        id="pass-number"
      >
        <i
          class="align-middle me-1 ri-check-line"
        />
        Um número (0-9)
      </li>
    </ul>
  </div>
</div>
`;
